packages/gatsby-plugin-sharp/src/index.js

```diff
@@ -614,13 +601,31 @@ async function fluid({ file, args = {}, reporter, cache }) {
     }
   }
 
+  // calculate presentationSizes
+  const maxWidth = options.maxWidth
+    ? Math.min(options.maxWidth, width)
+    : undefined
+  const maxHeight = options.maxHeight
+    ? Math.min(options.maxHeight, height)
+    : undefined
+  const imageWithDensity1 = images.find(
+    image => maxWidth === image.width || maxHeight === image.height
+  )
+  const presentationWidth = imageWithDensity1.width
+  const presentationHeight = imageWithDensity1.height
+
+  // If the users didn't set default sizes, we'll make one.
+  const sizes =
+    options.sizes ??
+    `(max-width: ${presentationWidth}px) 100vw, ${presentationHeight}px`
```

Comment:
```suggestion
    `(max-width: ${presentationWidth}px) 100vw, ${presentationWidth}px`
```
There is e2e gatsby-image test failing - not sure if this is reason for it, because failed assertion is:

> expected '' to match /\(max-width: \d+px\) 100vw, \d+px/

Reply:
maybe for some reason `options.sizes` is empty string here?